contracts/account-history-contract/src/types/account_snapshot_generator.rs

```diff
@@ -557,32 +538,36 @@ impl AccountSnapshotGenerator {
             self.metadata.usdc_apr_edenb.to_owned(),
             self.metadata.eden_apr_edenb.to_owned(),
         )
-        .map_or(
-            GetEdenBoostEarnProgramResp {
-                data: EdenBoostEarnProgram::default(),
-            },
-            |details| details,
-        );
+        .unwrap_or_default();
+
         let staked_asset_edenb = edenb_details.data;
-        total_balance = total_balance
+        total_staked_balance = total_staked_balance
```

Comment:
Lets remove this block entirely:

```
let staked_asset_edenb = edenb_details.data;
        total_staked_balance = total_staked_balance
            .checked_add(match staked_asset_edenb.clone() {
                EdenBoostEarnProgram {
                    rewards: Some(r), ..
                } => r.iter().fold(Decimal::zero(), |acc, item| {
                    acc.checked_add(item.usd_amount.map_or(Decimal::zero(), |amount| amount))
                        .map_or(Decimal::zero(), |res| res)
                }),
                _ => Decimal::zero(),
            }).unwrap_or_default();
        staked_assets.eden_boost_earn_program = staked_asset_edenb;
```

it's incorrectly adding eden boost rewards to the total staked balance, it's an old bug I believe. The reason we decided not to include rewards in this calculation is because they are added to the rewards calculation to show total portfolio balance later so this would count them twice.

Reply:
Here is the commit please have look

Reply:
Thank you